packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js

```diff
@@ -1676,39 +1688,250 @@ describe('plugin-meetings', () => {
 
-          let errorThrown = false;
+          const forceRtcMetricsSend = sinon.stub().resolves();
+          const closeMediaConnectionStub = sinon.stub();
+          Media.createMediaConnection = sinon.stub().returns({
+            close: closeMediaConnectionStub,
+            forceRtcMetricsSend,
+            getConnectionState: sinon.stub().returns(ConnectionState.Connected),
+            initiateOffer: sinon.stub().resolves({}),
+            on: sinon.stub(),
+          });
 
           await meeting
             .addMedia({
               mediaSettings: {},
             })
-            .catch(() => {
-              errorThrown = true;
+            .catch((err) => {
+              errorThrown = err;
+              assert.instanceOf(err, RetryWithTurnServerFailed);
             });
 
-          assert.calledOnceWithExactly(generateClientErrorCodeForIceFailureStub, {
+          assert.calledTwice(generateClientErrorCodeForIceFailureStub);
+          assert.calledWith(generateClientErrorCodeForIceFailureStub, {
+            signalingState: 'unknown',
+            iceConnectionState: 'unknown',
+            turnServerUsed: false,
+          })
+          assert.calledWith(generateClientErrorCodeForIceFailureStub, {
             signalingState: 'unknown',
             iceConnectionState: 'unknown',
             turnServerUsed: true,
           })
-          assert.calledOnceWithExactly(getErrorPayloadForClientErrorCodeStub, {clientErrorCode: MOCK_CLIENT_ERROR_CODE});
-          assert.calledTwice(webex.internal.newMetrics.submitClientEvent);
-          assert.calledWithMatch(webex.internal.newMetrics.submitClientEvent, {
+
+          assert.calledTwice(getErrorPayloadForClientErrorCodeStub);
+          assert.alwaysCalledWithExactly(getErrorPayloadForClientErrorCodeStub, {clientErrorCode: MOCK_CLIENT_ERROR_CODE});
+
+          assert.calledThrice(webex.internal.newMetrics.submitClientEvent);
+          assert.calledWith(webex.internal.newMetrics.submitClientEvent.firstCall,           {
+            name: 'client.media.capabilities',
+            payload: {
+              mediaCapabilities: {
+                rx: {
+                  audio: false,
+                  share: false,
+                  share_audio: false,
+                  video: false,
+                  whiteboard: false,
+                },
+                tx: {
+                  audio: false,
+                  share: false,
+                  share_audio: false,
+                  video: false,
+                  whiteboard: false,
+                },
+              },
+            },
+            options: {
+              meetingId: meeting.id,
+            },
+          });
+          assert.calledWith(webex.internal.newMetrics.submitClientEvent.secondCall, clientIceEndArgs);
+          assert.calledWith(webex.internal.newMetrics.submitClientEvent.thirdCall, clientIceEndArgs);
```

Comment:
I might be getting confused, but this is the check for the "client.ice.end" metric, right? I thought we would want to send it once and only if we fail both with and without TURN-TLS, or if we send it twice then the 2 events should be different: ICE_FAILED_WITHOUT_TURN_TLS_CLIENT_CODE vs ICE_FAILED_WITH_TURN_TLS_CLIENT_CODE.
I have a feeling that sending it twice will confuse the call analyzer, it feels like the fact that we're doing this retry is our own implementation detail and we should send the client.ice.end with canProceeed=false only once, after the retry fails.

Reply:
Sending it twice isn't ideal but doesn't confuse CA, See: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-480609 and the thread that I've linked in the description

I've changed it so that canProceed=true if turnServer wasn't used for that try, but as for not sending that CA event in excess that should be fixed as a part of the JIRA I just linked, not this PR